<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Mashware4017Class</class>
 <widget class="QMainWindow" name="Mashware4017Class">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>988</width>
    <height>718</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Mashware4017</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>../images/Marmoset Toolbag.ico</normaloff>../images/Marmoset Toolbag.ico</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="MyOpenGLWidget" name="openGLWidget">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>50</y>
      <width>841</width>
      <height>541</height>
     </rect>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>988</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>开始</string>
    </property>
    <addaction name="openFile"/>
    <addaction name="save"/>
   </widget>
   <widget class="QMenu" name="menu_2">
    <property name="title">
     <string>编辑</string>
    </property>
    <addaction name="moveUp"/>
    <addaction name="moveDown"/>
    <addaction name="moveRight"/>
    <addaction name="moveLeft"/>
    <addaction name="turnUp"/>
    <addaction name="turnDown"/>
    <addaction name="turnRight"/>
    <addaction name="turnLeft"/>
    <addaction name="magnify"/>
    <addaction name="narrow"/>
   </widget>
   <widget class="QMenu" name="menu_3">
    <property name="title">
     <string>帮助</string>
    </property>
    <addaction name="manual"/>
   </widget>
   <widget class="QMenu" name="menu_4">
    <property name="title">
     <string>显示</string>
    </property>
    <addaction name="fill"/>
    <addaction name="line"/>
   </widget>
   <widget class="QMenu" name="menu_5">
    <property name="title">
     <string>高级操作</string>
    </property>
    <addaction name="sparse"/>
    <addaction name="encrypt"/>
    <addaction name="smooth"/>
   </widget>
   <addaction name="menu"/>
   <addaction name="menu_2"/>
   <addaction name="menu_5"/>
   <addaction name="menu_4"/>
   <addaction name="menu_3"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>LeftToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="moveUp"/>
   <addaction name="moveDown"/>
   <addaction name="moveRight"/>
   <addaction name="moveLeft"/>
   <addaction name="separator"/>
   <addaction name="magnify"/>
   <addaction name="narrow"/>
   <addaction name="separator"/>
   <addaction name="turnUp"/>
   <addaction name="turnDown"/>
   <addaction name="turnLeft"/>
   <addaction name="turnRight"/>
   <addaction name="separator"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="openFile"/>
   <addaction name="save"/>
   <addaction name="separator"/>
   <addaction name="line"/>
   <addaction name="fill"/>
   <addaction name="separator"/>
   <addaction name="sparse"/>
   <addaction name="encrypt"/>
   <addaction name="smooth"/>
   <addaction name="separator"/>
   <addaction name="manual"/>
   <addaction name="separator"/>
  </widget>
  <action name="openFile">
   <property name="icon">
    <iconset>
     <normaloff>../images/171127-200.png</normaloff>../images/171127-200.png</iconset>
   </property>
   <property name="text">
    <string>打开</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="moveUp">
   <property name="icon">
    <iconset>
     <normaloff>../images/R-C (1).png</normaloff>../images/R-C (1).png</iconset>
   </property>
   <property name="text">
    <string>上移</string>
   </property>
   <property name="toolTip">
    <string>上移</string>
   </property>
   <property name="shortcut">
    <string>W</string>
   </property>
  </action>
  <action name="moveDown">
   <property name="icon">
    <iconset>
     <normaloff>../images/R-C.png</normaloff>../images/R-C.png</iconset>
   </property>
   <property name="text">
    <string>下移</string>
   </property>
   <property name="toolTip">
    <string>下移</string>
   </property>
   <property name="shortcut">
    <string>S</string>
   </property>
  </action>
  <action name="moveRight">
   <property name="icon">
    <iconset>
     <normaloff>../images/7dd98d1001e9390117e784b671ec54e737d19654.png</normaloff>../images/7dd98d1001e9390117e784b671ec54e737d19654.png</iconset>
   </property>
   <property name="text">
    <string>右移</string>
   </property>
   <property name="toolTip">
    <string>右移</string>
   </property>
   <property name="shortcut">
    <string>D</string>
   </property>
  </action>
  <action name="moveLeft">
   <property name="icon">
    <iconset>
     <normaloff>../images/R-C (2).png</normaloff>../images/R-C (2).png</iconset>
   </property>
   <property name="text">
    <string>左移</string>
   </property>
   <property name="toolTip">
    <string>左移</string>
   </property>
   <property name="shortcut">
    <string>A</string>
   </property>
  </action>
  <action name="turnUp">
   <property name="icon">
    <iconset>
     <normaloff>../images/s4.png</normaloff>../images/s4.png</iconset>
   </property>
   <property name="text">
    <string>向上翻转</string>
   </property>
   <property name="shortcut">
    <string>Alt+Up</string>
   </property>
  </action>
  <action name="turnDown">
   <property name="icon">
    <iconset>
     <normaloff>../images/1640074043897.png</normaloff>../images/1640074043897.png</iconset>
   </property>
   <property name="text">
    <string>向下翻转</string>
   </property>
   <property name="shortcut">
    <string>Alt+Down</string>
   </property>
  </action>
  <action name="turnRight">
   <property name="icon">
    <iconset>
     <normaloff>../images/1640074205253.png</normaloff>../images/1640074205253.png</iconset>
   </property>
   <property name="text">
    <string>向右翻转</string>
   </property>
   <property name="shortcut">
    <string>Alt+Right</string>
   </property>
  </action>
  <action name="turnLeft">
   <property name="icon">
    <iconset>
     <normaloff>../images/s6.png</normaloff>../images/s6.png</iconset>
   </property>
   <property name="text">
    <string>向左翻转</string>
   </property>
   <property name="toolTip">
    <string>向左翻转</string>
   </property>
   <property name="shortcut">
    <string>Alt+Left</string>
   </property>
  </action>
  <action name="fill">
   <property name="icon">
    <iconset>
     <normaloff>../images/fill.png</normaloff>../images/fill.png</iconset>
   </property>
   <property name="text">
    <string>填充显示</string>
   </property>
   <property name="toolTip">
    <string>填充显示</string>
   </property>
   <property name="shortcut">
    <string>F</string>
   </property>
  </action>
  <action name="manual">
   <property name="icon">
    <iconset>
     <normaloff>../images/manual.png</normaloff>../images/manual.png</iconset>
   </property>
   <property name="text">
    <string>使用指南</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="magnify">
   <property name="icon">
    <iconset>
     <normaloff>../images/images.png</normaloff>../images/images.png</iconset>
   </property>
   <property name="text">
    <string>放大</string>
   </property>
   <property name="toolTip">
    <string>放大</string>
   </property>
   <property name="shortcut">
    <string>Shift+Up</string>
   </property>
  </action>
  <action name="narrow">
   <property name="icon">
    <iconset>
     <normaloff>../images/sub.png</normaloff>../images/sub.png</iconset>
   </property>
   <property name="text">
    <string>缩小</string>
   </property>
   <property name="toolTip">
    <string>缩小</string>
   </property>
   <property name="shortcut">
    <string>Shift+Down</string>
   </property>
  </action>
  <action name="save">
   <property name="icon">
    <iconset>
     <normaloff>../images/save-1779882-1518534.png</normaloff>../images/save-1779882-1518534.png</iconset>
   </property>
   <property name="text">
    <string>保存</string>
   </property>
   <property name="toolTip">
    <string>保存</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="sparse">
   <property name="icon">
    <iconset>
     <normaloff>../images/clustering.png</normaloff>../images/clustering.png</iconset>
   </property>
   <property name="text">
    <string>稀疏</string>
   </property>
   <property name="toolTip">
    <string>稀疏</string>
   </property>
   <property name="shortcut">
    <string>Q</string>
   </property>
  </action>
  <action name="encrypt">
   <property name="icon">
    <iconset>
     <normaloff>../images/Cartoon-of-AMR-grids-with-two-levels-of-factor-2-refinement-The-coarsest-grid-covers-the.png</normaloff>../images/Cartoon-of-AMR-grids-with-two-levels-of-factor-2-refinement-The-coarsest-grid-covers-the.png</iconset>
   </property>
   <property name="text">
    <string>加密</string>
   </property>
   <property name="toolTip">
    <string>加密</string>
   </property>
   <property name="shortcut">
    <string>E</string>
   </property>
  </action>
  <action name="line">
   <property name="icon">
    <iconset>
     <normaloff>../images/R-C (3).png</normaloff>../images/R-C (3).png</iconset>
   </property>
   <property name="text">
    <string>线条显示</string>
   </property>
   <property name="toolTip">
    <string>线条显示</string>
   </property>
   <property name="shortcut">
    <string>L</string>
   </property>
  </action>
  <action name="smooth">
   <property name="icon">
    <iconset>
     <normaloff>../images/png-transparent-circle-white-circle-white-monochrome-black-thumbnail.png</normaloff>../images/png-transparent-circle-white-circle-white-monochrome-black-thumbnail.png</iconset>
   </property>
   <property name="text">
    <string>光滑</string>
   </property>
   <property name="toolTip">
    <string>光滑</string>
   </property>
   <property name="shortcut">
    <string>Z</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>MyOpenGLWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>myopenglwidget.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="Mashware4017.qrc"/>
 </resources>
 <connections/>
</ui>
